routemaster/state_machine/api.py

```diff
@@ -101,6 +101,37 @@ def create_label(app: App, label: LabelRef, metadata: Metadata) -> Metadata:
     return metadata
 
 
+def restore_label_and_restart(
+    app: App,
+    label: LabelRef,
+    replacement_metadata: Metadata,
+) -> Metadata:
+    """Restores a label that was previously deleted."""
```

Comment:
I'd probably extend this with a bit about why we're not merging the metadata like in the other API functions, but rather replacing entirely.

Reply:
496af12f6b37fabf61459b791a8ce8acfa625e83